solidity/test/mTestERC20.test.ts

```diff
@@ -0,0 +1,167 @@
+import { expect } from "chai"
+import { ethers } from "hardhat"
+import hre from "hardhat"
+import { loadFixture } from "@nomicfoundation/hardhat-network-helpers"
+import { getDeployedContract } from "./helpers/contract";
+
+describe("mTestERC20", function () {
+  const { deployments } = hre;
+  let mTestERC20: any;
+  let owner: any;
+  let governance: any;
+  let minter: any;
+  let user1: any;
+  let user2: any;
+
+  async function fixture() {
+    await deployments.fixture();
+    const signers = await ethers.getSigners();
+    owner = signers[0];
+    governance = signers[1];
+    minter = signers[2];
+    user1 = signers[3];
+    user2 = signers[4];
+
+    mTestERC20 = await getDeployedContract("mTestERC20");
+
+    return { mTestERC20, owner, governance, minter, user1, user2 }
+  }
+
+  describe("Initialization", function () {
+    it("should set the correct name and symbol", async function () {
+      const { mTestERC20 } = await loadFixture(fixture)
+      expect(await mTestERC20.name()).to.equal("Test Token")
+      expect(await mTestERC20.symbol()).to.equal("TEST")
+    })
+
+    it("should set the correct minter", async function () {
+      const { mTestERC20, minter } = await loadFixture(fixture)
+      expect(await mTestERC20.minter()).to.equal(minter)
+    })
+
+    it("should set the correct owner", async function () {
+      const { mTestERC20, owner } = await loadFixture(fixture)
+      expect(await mTestERC20.owner()).to.equal(owner.address)
+    })
```

Comment:
Should we add a test for setting decimals?

Reply:
We can 182fd47f8fb356e20071550a9f4906c50cee38b9